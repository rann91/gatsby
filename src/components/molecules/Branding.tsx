/* @jsx jsx */
import { Link } from 'gatsby'
import { Box, Flex, Image, jsx, Text } from 'theme-ui'
import useSiteMetadata from '../../hooks/use-site-metadata'
import logo from '../../images/logo.svg'
import { SiteSiteMetadata } from '../../typings/graphql'

interface Props {
  data: SiteSiteMetadata
}

export const PureBranding = ({ data: { title } }: Props) => (
  <Link sx={{ color: 'inherit', textDecoration: 'none' }} to="/">
    <Flex sx={{ alignItems: 'center' }}>
      <Box mr={3} sx={{ flexShrink: 0, lineHeight: 0 }}>
        <Image variant="logo" src={logo} alt="Branding Logo" />
      </Box>
      <Text variant="siteTitle">{title}</Text>
    </Flex>
  </Link>
)

const Branding = () => {
  const data = useSiteMetadata()

  return <PureBranding data={data} />
}

export default Branding
