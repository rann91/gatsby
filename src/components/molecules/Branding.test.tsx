import { render } from '@testing-library/react'
import React from 'react'
import { PureBranding as Branding } from './Branding'
import data from '../../json/metadata.json'

describe('Branding', () => {
  it('renders correctly', () => {
    const { container } = render(<Branding data={data} />)

    expect(container).toMatchSnapshot()
  })
})
