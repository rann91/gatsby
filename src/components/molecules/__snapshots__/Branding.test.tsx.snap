// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Branding renders correctly 1`] = `
<div>
  <a
    class="css-1fj3sfs-PureBranding"
    href="/"
  >
    <div
      class="css-tfpfm9-PureBranding"
    >
      <div
        class="css-7sc54i-PureBranding"
      >
        <img
          alt="Branding Logo"
          class="css-9taffg"
          src="test-file-stub"
        />
      </div>
      <div
        class="css-vurnku"
      >
        Gatsby
      </div>
    </div>
  </a>
</div>
`;
